index.js

```diff
@@ -73,12 +73,16 @@ export const Router = ({ hook, matcher, base = "", nested = false, children }) =
     router.hook = hook || proto.hook;
     router.matcher = matcher || proto.matcher;
     router.base = proto.base + base;
+
+    return router;
   };
```

Comment:
Can slightly improve efficiency here by pulling `updateRouter` out of the `Router` function.

E.g.
```typescript
const updateRouter = (router, hook, matcher, proto) => {...}

export const Router = (...) => {
  ...
  const [value] = useState(() => updateRouter({}, hook, matcher, proto));
  useIsomorphicLayoutEffect(() => {
      updateRouter(value, hook, matcher, proto);
  });
}

Reply:
Yep, but it kinda forces us to perform destructuring. I agree that there is a performance benefit, but at the same time we're loosing some points in stylistics terms. Plus, creating anonymous funcs in render [should be relatively cheap](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render)